src/lib/http.coffee

```diff
@@ -21,12 +23,84 @@ module.exports = (dnschain) ->
             @log = gNewLogger 'HTTP'
             @log.debug "Loading HTTPServer..."
             @rateLimiting = gConf.get 'rateLimiting:http'
+            app = express()
+
+            # Openname spec defined here:
+            # - https://github.com/okTurtles/openname-specifications/blob/resolvers/resolvers.md
+            # - https://github.com/openname/openname-specifications/blob/master/resolvers.md
+            
+            opennameRoute = express.Router()
+            
+            # Resolver specific API
+            opennameRoute.get /\/(?:resolver|dnschain)\/([^\/\.]+)(?:\.([a-z]+))?/, (req, res) =>
+                @log.debug gLineInfo("resolver API called"), {params: req.params}
+                [resource, format] = req.params
+                if resource == "fingerprint"
+                    if !format or format is 'json'
+                        res.json {fingerprint: @dnschain.encryptedserver.getFingerprint()}
+                    else
+                        @sendErr req, res, 400, "Unsupported format: #{format}"
+                else
+                    @sendErr req, res, 400, "Bad resource: #{resource}"
+
+            # Datastore API
+            opennameRoute.route(/// ^
+                \/(\w+)               # the datastore name
+                \/(\w+)               # the corresponding resource
+                (?:
+                    \/  ([^\/\.]+)    # optional property (or action on resource) 
+                    (?:
+                        \/ ([^\/\.]+) # optional action on property
+                    )?
+                )?
```

Comment:
@WeMeetAgain I think this regex can actually be written more elegantly.

This big part in the middle can, I think, be changed to:

```coffeescript
(?:\/([^\/\.]+))? # optional property (or action on resource)
(?:\/([^\/\.]+))? # optional action on property
```

You'll need to double-check that it produces the same results.